src/Sylius/Bundle/CoreBundle/Processor/ProductVariantCatalogPromotionsProcessor.php

```diff
@@ -46,15 +46,14 @@ public function process(ProductVariantInterface $variant): void
 
     private function reapplyOnChannelPricing(ChannelPricingInterface $channelPricing): void
     {
-        $appliedPromotions = $channelPricing->getAppliedPromotions();
+        $appliedPromotions = $channelPricing->getAppliedPromotions()->toArray();
         if (empty($appliedPromotions)) {
             return;
         }
-
         $this->catalogPromotionClearer->clearChannelPricing($channelPricing);
-        foreach ($appliedPromotions as $promotionCode => $promotionData) {
+        foreach ($appliedPromotions as $appliedPromotion) {
             /** @var CatalogPromotionInterface|null $catalogPromotion */
-            $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $promotionCode, 'enabled' => true]);
+            $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $appliedPromotion->getCode(), 'enabled' => true]);
```

Comment:
do you need it? you are iterating over an array with catalog promotion to get catalog promotion from the repository

Reply:
You're definitely right